src/utils/gh-auth.js

```diff
@@ -31,14 +31,26 @@ const promptForAuthMethod = async () => {
   return authMethod === authChoiceNetlify
 }
 
+/**
+ * Authenticate with the netlify app
+ * @returns {Promise<Record<string,string>} Returns a Promise with an object of the following shape
+ * ```
+ * {
+ *   user: 'spongebob,
+ *   token: 'gho_some-token',
+ *   provider: 'github'
+ * }
+ * ```
+ */
 const authWithNetlify = async () => {
   const port = await getPort({ port: SERVER_PORT })
   const { promise: deferredPromise, reject: deferredReject, resolve: deferredResolve } = createDeferred()
 
   const server = http.createServer(function onRequest(req, res) {
     const parameters = new URLSearchParams(req.url.slice(req.url.indexOf('?') + 1))
     if (parameters.get('token')) {
-      deferredResolve(parameters)
+      const recordOfParams = [...parameters].reduce((prev, [key, value]) => ({ ...prev, [key]: value }), {})
```

Comment:
@lukasholzer While going through the fix for this bug yesterday, I found out that the `cli` repo already has a polyfill for `fromEntries` (which is what we're trying to do here).

You can use it like this:
`const fromEntries = require('@ungap/from-entries')`

`const recordOfParams = fromEntries([...parameters])`

Reply:
thx for pointing out :)